src/components/bluetooth-devices/bluetooth-devices.tsx

```diff
@@ -0,0 +1,129 @@
+import { BLE } from '../../core/wallet/hw-wallet/ledger/transport/transport-ble';
+
+export interface IExternalProps {
+    obRef: any;
+    onComplete: (deviceId: string) => any;
+}
+
+interface IState {
+    devices: [];
+    isVisible: boolean;
+    error: Error;
+    refreshing: boolean;
+}
+
+const deviceAddition = device => ({ devices }) => ({
+    devices: devices.some(i => i.id === device.id) ? devices : devices.concat(device)
+});
+
+export class BluetoothDevicesModalComponent extends React.Component<
+    IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    public scannerUnsubscribe = null;
+    constructor(props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>) {
+        super(props);
+
+        this.state = {
+            devices: [],
+            isVisible: false,
+            error: null,
+            refreshing: false
+        };
+        props.obRef && props.obRef(this);
+    }
+
+    public open = async () => {
+        this.setState({ isVisible: true });
+
+        this.scannerUnsubscribe = await BLE.scan(event => {
+            this.setState(deviceAddition(event.data));
+        });
+    };
+
+    public renderItem = item => {
+        return (
+            <TouchableOpacity
+                style={this.props.styles.deviceRow}
+                onPress={async () => {
+                    try {
+                        await BLE.connect(item.id);
+                        this.scannerUnsubscribe.unsubscribe();
+                        this.props.onComplete(item.id);
+                        this.setState({ isVisible: false });
+                    } catch (error) {
+                        // show error
+                    }
+                }}
+            >
+                <Text style={this.props.styles.text}>{item.name}</Text>
+            </TouchableOpacity>
+        );
+    };
+
+    public renderHeader = () => {
+        const styles = this.props.styles;
+        const { error } = this.state;
+        return error ? (
+            <View style={styles.header}>
+                <Text style={styles.headerTitle}>Sorry, an error occured</Text>
+                <Text style={styles.errorTitle}>{String(error.message)}</Text>
+            </View>
+        ) : (
+            <View style={styles.header}>
+                <TouchableOpacity
+                    testID="button-close"
+                    style={styles.close}
+                    onPress={() => {
+                        // unsubscribe
+                        this.scannerUnsubscribe.unsubscribe();
+                        this.setState({ isVisible: false });
+                    }}
+                >
+                    <Icon name="close" size={ICON_SIZE} style={styles.icon} />
+                </TouchableOpacity>
+                <Text style={styles.headerTitle}>Scanning for Bluetooth...</Text>
+                <View style={styles.indicator}>
+                    <ActivityIndicator size="large" color="#ffffff" />
```

Comment:
use loader component done by @NovacAndrei instead

Reply:
i didnt used it as final design. - updated